pkg/virtual/framework/wrappers/rbac/merging.go

```diff
@@ -0,0 +1,436 @@
+	aggregatedInformers := make([]rbacinformers.ClusterRoleBindingInformer, 0)
+	for _, inf := range i.informers {
+		aggregatedInformers = append(aggregatedInformers, inf.ClusterRoleBindings())
+	}
+	return MergedClusterRoleBindingInformer(aggregatedInformers)
+}
+
+func (i *mergedInterface) ClusterRoles() rbacinformers.ClusterRoleInformer {
+	aggregatedInformers := make([]rbacinformers.ClusterRoleInformer, 0)
+	for _, inf := range i.informers {
+		aggregatedInformers = append(aggregatedInformers, inf.ClusterRoles())
+	}
+	return MergedClusterRoleInformer(aggregatedInformers)
+}
+
+func (i *mergedInterface) RoleBindings() rbacinformers.RoleBindingInformer {
+	aggregatedInformers := make([]rbacinformers.RoleBindingInformer, 0)
+	for _, inf := range i.informers {
+		aggregatedInformers = append(aggregatedInformers, inf.RoleBindings())
+	}
+	return MergedRoleBindingInformer(aggregatedInformers)
+}
+
+func (i *mergedInterface) Roles() rbacinformers.RoleInformer {
+	aggregatedInformers := make([]rbacinformers.RoleInformer, 0)
+	for _, inf := range i.informers {
+		aggregatedInformers = append(aggregatedInformers, inf.Roles())
+	}
+	return MergedRoleInformer(aggregatedInformers)
+}
+
+func MergedClusterRoleBindingInformer(informers []rbacinformers.ClusterRoleBindingInformer) rbacinformers.ClusterRoleBindingInformer {
+	return &mergedClusterRoleBindingInformer{
+		informers: informers,
+	}
+}
+
+var _ rbacinformers.ClusterRoleBindingInformer = (*mergedClusterRoleBindingInformer)(nil)
+var _ rbaclisters.ClusterRoleBindingLister = (*mergedClusterRoleBindingLister)(nil)
+
+type mergedClusterRoleBindingInformer struct {
+	informers []rbacinformers.ClusterRoleBindingInformer
+}
+
+type mergedClusterRoleBindingLister struct {
+	listers []rbaclisters.ClusterRoleBindingLister
+}
+
+func (i *mergedClusterRoleBindingInformer) Informer() cache.SharedIndexInformer {
+	panic("not implemented")
+}
+
+func (i *mergedClusterRoleBindingInformer) Lister() rbaclisters.ClusterRoleBindingLister {
+	aggregatedListers := make([]rbaclisters.ClusterRoleBindingLister, 0)
+	for _, inf := range i.informers {
+		aggregatedListers = append(aggregatedListers, inf.Lister())
+	}
+	return &mergedClusterRoleBindingLister{
+		listers: aggregatedListers,
+	}
+}
+
+func (l *mergedClusterRoleBindingLister) List(selector labels.Selector) (ret []*rbacv1.ClusterRoleBinding, err error) {
+	aggregatedList := make([]*rbacv1.ClusterRoleBinding, 0)
+	for _, lister := range l.listers {
+		list, err := lister.List(selector)
+		if err != nil {
+			return nil, err
+		}
+		aggregatedList = append(aggregatedList, list...)
+	}
+	return aggregatedList, nil
+}
+
+func (l *mergedClusterRoleBindingLister) Get(name string) (*rbacv1.ClusterRoleBinding, error) {
+	var errorHolder error
+	for _, lister := range l.listers {
+		item, err := lister.Get(name)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				errorHolder = err
+			}
+		} else {
+			return item, nil
```

Comment:
have you experimented with merging the object by concatenating the rules?

Reply:
No. Could you please give me an example how this should work?

Reply:
(of course in the ClusterRole case, not here with the binding).

Just return an new object there that includes the concatenation of the rules, but the same name.

Reply:
aaah, yes, that makes a lot of sense. Thanks for pointing this out!